go/tools/gazelle/resolve/resolve_external.go

```diff
@@ -117,6 +117,22 @@ func (r *externalResolver) lookupPrefix(importpath string) (string, error) {
 		subpaths = append(subpaths, prefix)
 	}
 
+	// gopkg.in is special, and might have either one or two levels of
+	// missing paths.
+	//
+	// https://github.com/bazelbuild/rules_go/issues/998
+	if strings.HasPrefix(importpath, "gopkg.in/") {
```

Comment:
Since you're using regexp, might as well use it to match the whole string. Something like `^(gopkg.in/(?:[^/]+/)?[^/]+\.v\d+)(?:/|$)` should work (haven't tested, but hopefully that's close).

Make sure to compile the regexp in global scope.

Reply:
Done.

That regex worked as-is.